### PR TITLE
feat: enable generic kernel (hwe)

### PR DESCRIPTION
**Pull request by yanksyoon**
Applicable spec: N/A

### Overview

- Installs HWE kernel.

### Rationale

- To match parity with GitHub hosted runners.
- 
### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The documentation is generated using `src-docs`
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
- [x] The application version in `pyproject.toml` is incremented

### CODE REVIEW

**Review by arturo-seijas — APPROVED** [review]

File: src/github_runner_image_builder/config.py [hunk 1]
```diff
@@ -102,6 +102,24 @@ class BaseImage(str, Enum):
...
+        Return:
+            The release version of the current base image.
+        """
+        match base:
+            case BaseImage.JAMMY:
+                return "22.04"
+            case BaseImage.NOBLE:
+                return "24.04"
+        # This case is not possible
+        return "22.04"  # pragma: no cover
```
Comment (cbartz) [inline src/github_runner_image_builder/config.py:121]: Could we not remove this code and instead ignore another linting error (e.g. probably one for the typehint)?

File: tests/unit/test_config.py [hunk 2]
```diff
@@ -110,3 +110,19 @@ def test_base_image(image: str, expected_base_image: BaseImage):
     assert: expected base image is returned.
     """
     assert BaseImage.from_str(image) == expected_base_image
+
+
+@pytest.mark.parametrize(
+    "base_image, expected_version",
+    [
+        pytest.param(BaseImage.JAMMY, "22.04", id="jammy"),
+        pytest.param(BaseImage.NOBLE, "24.04", id="22.04"),
```
Comment (cbartz) [inline tests/unit/test_config.py:119]: The id does not look right here.

**Review by cbartz — COMMENTED** [review]
Looks good, some nitpicks

File: src/github_runner_image_builder/config.py (hunk 1 above)
Comment (yanksyoon) [reply src/github_runner_image_builder/config.py:121]: Would you mean we should remove this code and ignore the lint error?

File: tests/unit/test_config.py (hunk 2 above)
Comment (yanksyoon) [reply tests/unit/test_config.py:119]: You must be a good fisher with all these great catches!

File: src/github_runner_image_builder/config.py (hunk 1 above)
Comment (yanksyoon) [reply src/github_runner_image_builder/config.py]: Removed!

**Review by cbartz — APPROVED** [review]

**Review by javierdelapuente — APPROVED** [review]

**Pull request merged**